hack/.ci/lib/e2e.sh

```diff
@@ -69,7 +75,31 @@ EOF
 }
 
 function gather-artifact-on-exit {
-  gather-artifacts || "Error gathering artifacts" > /dev/stderr
+  gather-artifacts "${ARTIFACTS}/must-gather" || "Error gathering artifacts" > /dev/stderr
```

Comment:
Why isn't gathering the artifacts in every cluster hooked up by default here? if there is more then one config you gather more if there is only a kubeconfig you still do the same thing we did so far.

Reply:
Because the other script (release) doesn't support it for now (at least this PR doesn't introduce support there), and imo at this point we would have to start doing the same thing for other functions, so we start losing a degree of control. https://github.com/scylladb/scylla-operator/pull/1881#discussion_r1622273827

Reply:
I meant it more like to auto-detect the kubeconfigs variables but this is a good start.

Reply:
I wanted to do that at first but found this more readable. It also allows to pass a subset if needed.